semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/MD5.java

```diff
@@ -0,0 +1,35 @@
+package com.sourcegraph.semanticdb_javac;
+
+import java.nio.CharBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/** Utility to compute MD5 checksums of strings. */
+public final class MD5 {
+  private static final char[] HEX_ARRAY;
+
+  static {
+    HEX_ARRAY = "0123456789ABCDEF".toCharArray();
+  }
+
+  private static String bytesToHex(byte[] bytes) {
+    char[] hexChars = new char[bytes.length * 2];
+    int j = 0;
+    while (j < bytes.length) {
+      int v = bytes[j] & 0xFF;
+      hexChars[j * 2] = HEX_ARRAY[v >>> 4];
+      hexChars[j * 2 + 1] = HEX_ARRAY[v & 0x0F];
+      j += 1;
+    }
+    return new String(hexChars);
+  }
```

Comment:
I _think_ this could be replaced with [javax.xml.bind.DataTypeConverter.printHexBinary](https://docs.oracle.com/javase/8/docs/api/javax/xml/bind/DatatypeConverter.html#printHexBinary-byte:A-) but no big deal

Reply:
IntelliJ says this class exists in `rt.jar` but I can't find it in my Java home with

![Screenshot 2021-02-11 at 14 10 11](https://user-images.githubusercontent.com/1408093/107640642-e04ac480-6c72-11eb-9e23-5c6e795229cb.png)

```
❯ jar tf jre/lib/rt.jar  | grep javax.xml.bind.DataTypeConverter
<empty>
❯ javap -cp jre/lib/rt.jar javax.xml.bind.DataTypeConverter
Error: class not found: javax.xml.bind.DataTypeConverter
❯ javap -cp lib/tools.jar javax.xml.bind.DataTypeConverter
Error: class not found: javax.xml.bind.DataTypeConverter
```